chainntnfs/txconfnotifier.go

```diff
@@ -370,16 +593,24 @@ out:
 	// this new height.
 	for _, txHashes := range tcn.txsByInitialHeight {
 		for txHash := range txHashes {
-			for _, ntfn := range tcn.confNotifications[txHash] {
-				// If the notification hasn't learned about the
-				// confirmation of its transaction yet (in the
-				// case of historical confirmations), we'll skip
-				// it.
-				if ntfn.details == nil {
-					continue
-				}
+			confSet, ok := tcn.confNotifications[txHash]
+			if !ok {
+				Log.Warnf("Txns by initial height index "+
+					"contains key not known to "+
+					"confNotifications: %v", txHash)
+				continue
+			}
+
+			// If the confset hasn't learned about the confirmation
+			// of its transaction yet, we can't schedule the
+			// dispatch.
+			if confSet.details == nil {
```

Comment:
I think this is also an error?

Reply:
removed in favor of panic on dereference